### ISSUE TITLE
100% CPU Usage - Runs out of memory

### ISSUE THREAD

**Issue by tflanagan** [npm/npm#13393]
#### I'm opening this issue because:
- [x] npm is crashing.
- [x] npm is doing something I don't understand.
#### What's going wrong?

I'm running Windows 10 with a fresh Nodejs installation. Nodejs itself works fine, but NPM causes 100% cpu usage, consumed memory continues to grow until process runs out of memory.

```
$ node --version
v4.4.7

$ npm --version
3.6.0

$ npm

<--- Last few GCs --->

   41044 ms: Scavenge 1401.7 (1457.9) -> 1401.7 (1457.9) MB, 7.4 / 0 ms (+ 2.0 ms in 1 steps since last GC) [allocation failure] [incremental marking delaying mark-sweep].
   41863 ms: Mark-sweep 1401.7 (1457.9) -> 1401.7 (1457.9) MB, 818.5 / 0 ms (+ 2.0 ms in 2 steps since start of marking, biggest step 2.0 ms) [last resort gc].
   42683 ms: Mark-sweep 1401.7 (1457.9) -> 1401.7 (1457.9) MB, 820.0 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 00000018636B4639 <JS Object>
    1: ToPropertyDescriptor(aka ToPropertyDescriptor) [native v8natives.js:~234] [pc=00000316658207A8] (this=00000018636041B9 <undefined>,J=000000DEFCCC2A21 <an Object with map 000002F076791319>)
    2: defineProperty(aka defineProperty) [native v8natives.js:775] [pc=000003166564A934] (this=00000018636041B9 <undefined>,J=000000DEFCCC2AD9 <an Error with map 0000002825008A91>,V=00000018636350A9...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory

$ npm config get registry

<--- Last few GCs --->

   38387 ms: Scavenge 1402.0 (1457.9) -> 1402.0 (1457.9) MB, 7.8 / 0 ms (+ 2.0 ms in 1 steps since last GC) [allocation failure] [incremental marking delaying mark-sweep].
   39242 ms: Mark-sweep 1402.0 (1457.9) -> 1402.0 (1457.9) MB, 854.8 / 0 ms (+ 3.0 ms in 2 steps since start of marking, biggest step 2.0 ms) [last resort gc].
   40063 ms: Mark-sweep 1402.0 (1457.9) -> 1402.0 (1457.9) MB, 821.4 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0000029B80CB4639 <JS Object>
    1: ToPropertyDescriptor(aka ToPropertyDescriptor) [native v8natives.js:~234] [pc=00000145D2020288] (this=0000029B80C041B9 <undefined>,J=000003A9154C2A31 <an Object with map 0000003FEAC912C1>)
    2: defineProperty(aka defineProperty) [native v8natives.js:775] [pc=00000145D1E4A934] (this=0000029B80C041B9 <undefined>,J=000003A9154C2AE9 <an Error with map 00000237D1108A91>,V=0000029B80C350A9...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```

**Comment by Tanax** [comment]
I get the same on Mac when npm is building its local index for the first time.

**Comment by KenanY** [comment]
@Tanax `npm search` was already fixed in `npm@4`.

**Comment by Tanax** [comment]
@KenanY Cheers. Worked. Let's hope it gets into the latest release soon :)

**Comment by legodude17** [comment]
@KenanY, can this be closed? Is `npm@4` `latest` yet.

**Comment by tflanagan** [comment]
@legodude17 

Depends, if I wipe my computer again and reinstall node/npm, will I still run into this issue?

**Comment by legodude17** [comment]
If you install `npm@4`. 😄

**Comment by tflanagan** [comment]
I'm not sure if you have anything LTS-esque, but that answer isn't reassuring...

**Comment by legodude17** [comment]
What do you mean? The change to search is technically a breaking change, so it is only in the new major version.

**Comment by KenanY** [comment]
Currently, `npm@3.10.9` is what's tagged as the latest stable version. Also, to me, it looks like @tflanagan's issue is unrelated to `npm search`, so I don't think this can be closed yet.

**Comment by legodude17** [comment]
Why do you think it is not related @KenanY?

**Comment by KenanY** [comment]
@legodude17 `npm search` was failing because it would try to load tons of registry data into memory at once. This was fixed by streaming in the data instead. Here, just running `npm` alone seems to be causing it to run out of memory, which seems very odd to me and _probably_ not affected by the `npm search` fix.

**Comment by legodude17** [comment]
Oh, you are right, confused @Tanax's issue for @tflanagan. How much RAM do you have @tflanagan?

**Comment by tflanagan** [comment]
16GB

**Comment by legodude17** [comment]
That should be plenty. Try making a quick node program to use a bit of RAM and test that. Like try fetching a youtube video or some music. 😄

**Comment by tflanagan** [comment]
I've since wiped and reinstalled a higher version of nodejs and npm, so I cannot test.

This was on a fresh (updated) windows 10 installation, with a fresh version of Nodejs and NPM at the versions in the first post.

**Comment by legodude17** [comment]
Does the error happen now? With latest versions?

**Comment by helio-frota** [comment]
This can be closed ?

**Comment by tflanagan** [comment]
I went to uninstall an old version of Nodejs then install a newer version, typical installation process, nothing fancy. Opened cmder, went to check and ran into it again.

This time it took quite a while. The process plateaued at 1.4 - 1.5 gb of ram, but cpu was at a consistent 90 - 100%. I was able to recognize the situation, find this issue, type this reply up, and the process was still running. I went to grab a beer, came back, still running, still using the same resources. As I grew bored, I `ctrl+c`'d it.

Attempting to debug this, I saw some npm files survived the uninstall. `npm-cache` in roaming, deleted this folder, tried again, same issue.

`npm` in roaming survived as well. Deleting the `npm` folder in roaming removed this issue, unsure if `npm-cache` has anything to do with it.
